### PR TITLE
Add last activity tracker for Project & User

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- #1291
- #1292
- #1293

## Changes

* Add the last read/write activity to Project
* Add the last active date/status to the user using the notification node

> NOTE: Using Redis to track data and using Celery to push track data to the database every 6 hours

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/project/tests/test_schemas.py [hunk 1]
```diff
@@ -865,6 +870,134 @@ def _expected_response(
...
+        for project in project_with_access:
+            project.add_member(user, role=self.project_role_member)
+
+        def _query_check(project_id, **kwargs):
+            return self.query_check(MUTATION, variables={'projectId': project_id}, **kwargs)
+
+        self.force_login(user)
+
+        with patch('deep.trackers.timezone.now') as timezone_now_mock:
+            timezone_now_mock.return_value = timezone_now = datetime(2021, 1, 1, 0, 0, 0, 123456, tzinfo=pytz.UTC)
```
Comment (bewakes) [inline apps/project/tests/test_schemas.py]: Please add notes on what this test case is trying to achieve.

File: deep/trackers.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    # -- Write
+    update_entity_data_in_bulk(
+        Project,
+        CacheKey.Tracker.LAST_PROJECT_WRITE_ACCESS_DATETIME,
+        'last_write_access',
+    )
+    # -- Update project->status using last_write_access
+    # -- -- To active
+    Project.objects.filter(
+        last_write_access__lte=timezone.now() - relativedelta(years=1),
```
Comment (bewakes) [inline deep/trackers.py]: Maybe we can have a constant like `PROJECT_INACTIVE_AFTER_YEARS` or something like that.

File: deep/trackers.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+def update_user_data_in_bulk():
+    update_entity_data_in_bulk(
+        Profile,
+        CacheKey.Tracker.LAST_USER_ACTIVE_DATETIME,
+        'last_active',
+    )
+    # -- Update user->profile-is_active using last_active
+    # -- -- To active
+    Profile.objects.filter(
+        last_active__lte=timezone.now() - relativedelta(months=6),
```
Comment (bewakes) [inline deep/trackers.py]: Same here, maybe a const for 6.

**Review by bewakes — CHANGES_REQUESTED** [review]

File: apps/project/tests/test_schemas.py (hunk 1 above)
Comment (thenav56) [reply apps/project/tests/test_schemas.py]: Done https://github.com/the-deep/server/commit/3a19f3fd39d0d5d09253f94d32acdb1c4db0f953

File: deep/trackers.py (hunk 2 above)
Comment (thenav56) [reply deep/trackers.py]: Done https://github.com/the-deep/server/commit/3a19f3fd39d0d5d09253f94d32acdb1c4db0f953

File: deep/trackers.py (hunk 3 above)
Comment (thenav56) [reply deep/trackers.py]: Done https://github.com/the-deep/server/commit/3a19f3fd39d0d5d09253f94d32acdb1c4db0f953

**Review by bewakes — APPROVED** [review]

File: apps/project/models.py [hunk 4]
```diff
@@ -43,6 +43,8 @@ class Status(models.TextChoices):
         ACTIVE = 'active', 'Active'
         INACTIVE = 'inactive', 'Inactive'
 
+    PROJECT_INACTIVE_AFTER_MONTHS = 12
```
Comment (k9845) [inline apps/project/models.py:46]: Whynot store in `settings.py`?

**Review by k9845 — APPROVED** [review]

File: apps/project/models.py (hunk 4 above)
Comment (thenav56) [reply apps/project/models.py:46]: For now, this is constant.

**Pull request merged**